libmamba/src/core/subdirdata.cpp

```diff
@@ -61,7 +61,7 @@ namespace mamba
 
         for (const auto& c : Context::instance().repodata_has_zst)
         {
-            if (channel->channel_context().make_channel(c) == *channel)
+            if (contains(c, channel->location()) && contains(c, channel->name()))
```

Comment:
@wolfv I am wondering if the general common cases of using `repodata_has_zst` are mostly urls, and so changing this line to just checking the `name` and `location` of the channel against the configured `repodata_has_zst` would be enough? (This would clearly simplify linking the channel_context to a channel...)